module_utils/storage_lsr/argument_validator.py

```diff
@@ -0,0 +1,171 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+class ArgValidator(object):
+
+    @classmethod
+    def _validate_list(cls, value):
+        if isinstance(value, list):
+            return value
+
+        if isinstance(value, str):
+            return value.split(",")
+
+        if isinstance(value, int) or isinstance(value, float):
+            return [value]
+
+        raise TypeError("'%s' has to be a list" % value)
+
+    @classmethod
+    def _validate_str(cls, value):
+        return str(value)
+
+    @classmethod
+    def _validate_bool(cls, value):
+        BOOLEANS_TRUE = ['y', 'yes', 'on', '1', 'true', 't', 1, 1.0, True]
+        BOOLEANS_FALSE = ['n', 'no', 'off', '0', 'false', 'f', 0, 0.0, False]
+
+        if value in BOOLEANS_TRUE:
+            return True
+        if value in BOOLEANS_FALSE:
+            return False
+        if value is None:
+            return None
+        raise TypeError("'%s' has to be convertable to a bool" % value)
+
+    @classmethod
+    def _validate_int(cls, value):
+        if isinstance(value, int):
+            return value
+        if isinstance(value, str):
```

Comment:
see other comment about strings
also, can't you just do something like this?:
```

    def _validate_int(cls, value):
        try:
            return int(value)
        except ValueError:
            pass
        raise TypeError("'%s' has to be an integer" % value)
```

Reply:
Certainly much more elegant solution. Fixed.